optuna/storages/_base.py

```diff
@@ -556,7 +555,11 @@ def get_n_trials(self, study_id: int, state: Optional[TrialState] = None) -> int
             :exc:`KeyError`:
                 If no study with the matching ``study_id`` exists.
         """
-        raise NotImplementedError
+        self.read_trials_from_remote_storage(study_id)
+        if state is None:
+            return len(self.get_all_trials(study_id, deepcopy=False))
+
+        return len([t for t in self.get_all_trials(study_id, deepcopy=False) if t.state == state])
```

Comment:
The logic looks good, but do you think we should add tests for the `state` parameter? Seems like it's not tested, going through `test_storages.py`.